docs/resources/client/models/modelloaders.md

```diff
@@ -392,65 +237,80 @@ public class MyLoaderBuilder extends CustomLoaderBuilder<BlockModelBuilder> {
+// This assumes an extension of ModelProvider and a DeferredBlock<Block> EXAMPLE_BLOCK.
+// The parameter for customLoader() is a Supplier to construct the builder and a Consumer to set to associated properties.
+@Override
+protected void registerModels(BlockModelGenerators blockModels, ItemModelGenerators itemModels) {
+    blockModels.createTrivialBlock(
+        // The block to generate the model for
+        EXAMPLE_BLOCK.get(),
+        TexturedModel.createDefault(
+            // A mapping used to get the textures
+            block -> new TextureMapping().put(
+                TextureSlot.ALL, TextureMapping.getBlockTexture(block)
+            ),
+            // The model template builder used to create the JSON
+            ExtendedModelTemplateBuilder.builder()
+                // Say we are using a custom model loader
+                .customLoader(MyLoaderBuilder::new, loader -> {
+                    // Set any required fields here
+                })
+                // Textures required by the model
+                .requiredTextureSlot(TextureSlot.ALL)
+                // Call build once complete
+                .build()
+        )
+    )
+}
 ```
 
-Then, call your field setters on the `loaderBuilder`.
-
 #### Visibility
 
-The default implementation of `CustomLoaderBuilder` holds methods for applying visibility. You may choose to use or ignore the `visibility` property in your model loader. Currently, only the [composite model loader][composite] makes use of this property.
+The default implementation of `CustomLoaderBuilder` holds methods for applying visibility. You may choose to use or ignore the `visibility` property in your model loader. Currently, only the [composite model loader][composite] and [OBJ loader][obj] makes use of this property.
 
 ### Reusing the Default Model Loader
 
-In some contexts, it makes sense to reuse the vanilla model loader and just building your model logic on top of that instead of outright replacing it. We can do so using a neat trick: In the model loader, we simply remove the `loader` property and send it back to the model deserializer, tricking it into thinking that it is a regular model now. We then pass it to the geometry, bake the model geometry there (like the default geometry handler would) and pass it along to the dynamic model, where we can then use the model's quads in whatever way we want:
+In some contexts, it makes sense to reuse the vanilla model loader and just building your model logic on top of that instead of outright replacing it. We can do so using a neat trick: in the model loader,  we simply remove the `loader` property and send it back to the model deserializer, tricking it into thinking that it is a regular unbaked model now. Then, we bake the model during the baking process to pass along to the baked model, where we can the use the model's quads in whatever way we want:
```

Comment:
This section should mention that the example below only applies if the entire file itself with the loader key removed or a single nested model is loaded again. In cases similar to the composite model, `DelegateBakedModel` should not be used and the inner models should instead be baked via `UnbakedModel.bakeWithTopLevelValues()`.